lib/webaudio/audiocontext.js

```diff
@@ -1,16 +1,79 @@
 /* globals webkitAudioContext, AudioContext */
 'use strict';
 
-var NativeAudioContext = null;
+/**
+ * @interface AudioContextFactoryOptions
+ * @property {AudioContext} [AudioContext] - The AudioContext constructor
+ */
 
-if (typeof AudioContext !== 'undefined') {
-  NativeAudioContext = AudioContext;
-} else if (typeof webkitAudioContext !== 'undefined') {
-  NativeAudioContext = webkitAudioContext;
+/**
+ * {@link AudioContextFactory} ensures we construct at most one AudioContext
+ * at a time, and that it is eventually closed when we no longer need it.
+ * @param {AudioContextFactoryOptions} [options]
+ * @property {AudioContextFactory} AudioContextFactory - The
+ *   {@link AudioContextFactory} constructor
+ */
+function AudioContextFactory(options) {
+  var NativeAudioContext = options && options.AudioContext
```

Comment:
If we move the binding `NativeAudioContext` out, what would you want to name the binding inside the constructor? Because we still need to choose between the native `AudioContext` and the one passed in via `options`